### PR TITLE
Custom rebirth npc ID

### LINKED ISSUE #30: Custom NPC scripts don't override the default
If you create a custom NPC script, it isn't executed if the NPC already has a `D0` entry in the wz xml.

For example, Cody has the following XML:
```
  <imgdir name="9200000">
    <string name="name" value="Cody"/>
    <string name="func" value="Wizet Wizard"/>
    <string name="n0" value="Hey, care to help me out??"/>
    <string name="n1" value="I&apos;m telling you, I really AM from WIZET!"/>
    <string name="d0" value="What&apos;s going on? I&apos;m Cody, the head programmer of MapleStory~"/>
    <string name="d1" value="Thanks to your help, the party went bananas! Boy it was fun. anyway, hope you liked the stuff that I gave you. I&apos;ll see you around~"/>
  </imgdir>
```
I created a new npc file (`scripts/npc/9200000.js`) and gave it custom content, but it's never executed. It seems most if not all of the npcs that are already in the WZ xml files have been skipped over if they already have text. It makes sense, but it's difficult if you want to have an npc become a custom npc.

I think the scripts should take precedence if they exist, instead of being a fallback. I think `scipting.npc.NPCConversationManager` may be the place to do it, but I haven't dug far enough to say for sure. 

A side use-case for this is fixing incorrect wording that was on some of the NPCs in v83.

Thoughts?

P.s. I came across this issue when trying to debug why Cody wasn't registering as an npc when debug mode is enabled and I clicked him. Seems like there are multiple pathways to having a conversation with an npc, maybe something to refactor.

### LINKED ISSUE #32: Change rebirth support to allow a custom npc id to be used
Currently rebirths can be turned on but they'll only we usable if you can find `Wolf Spirit Ryko` (9010021). We couldn't find that npc anywhere that was listed in the xml files, so I modified the rebirth system to allow a specific NPC id to be passed in.

This feature seems pretty useful, so I think it should be added. 

Since npcs that already have their conversations present in the `wz` don't automatically use the `*.js` file present and any npc should be usable, the id of the rebirth npc being set should be added to the list of npcs that will always use the override, if rebirths are enabled.

### PR DESCRIPTION
**Pull request by wejrox**
Closes #32 , closes #30

### CODE REVIEW

File: src/main/java/net/server/channel/handlers/PlayerLoggedinHandler.java [hunk 1]
```diff
@@ -409,7 +408,14 @@ public final void handlePacket(SeekableLittleEndianAccessor slea, MapleClient c)
                 }
                 
                 if (YamlConfig.config.server.USE_NPCS_SCRIPTABLE) {
-                    c.announce(MaplePacketCreator.setNPCScriptable(ScriptableNPCConstants.SCRIPTABLE_NPCS));
+                    List<Integer> npcsIds = YamlConfig.config.server.NPCS_SCRIPTABLE;
+
+                    // Any npc be specified as the rebirth npc. Allow the npc to use custom scripts explicitly.
+                    if (YamlConfig.config.server.USE_REBIRTH_SYSTEM) {
+                        npcsIds.add(YamlConfig.config.server.REBIRTH_NPC_ID);
+                    }
```
Comment (P0nk) [inline src/main/java/net/server/channel/handlers/PlayerLoggedinHandler.java:416]: You are mutating the global list of npcsIds. Each time a player logs in and the rebirth system is enabled, the rebirth npc id will be appended, so after a while, when a bunch of players have logged in, there will be a bunch of duplicate npc ids at the end of the list.
You should do something like:
`List<Integer> npcsIds = new ArrayList(YamlConfig.config.server.NPCS_SCRIPTABLE);` on line 411

File: src/main/java/tools/MaplePacketCreator.java [hunk 2]
```diff
@@ -8318,17 +8318,24 @@ public static void addCashItemInformation(final MaplePacketLittleEndianWriter mp
...
                 MaplePacketLittleEndianWriter mplew = new MaplePacketLittleEndianWriter();
                 mplew.writeShort(SendOpcode.SET_NPC_SCRIPTABLE.getValue());
-                mplew.write(scriptNpcDescriptions.size());
-                for (Pair<Integer, String> p : scriptNpcDescriptions) {
-                        mplew.writeInt(p.getLeft());
-                        mplew.writeMapleAsciiString(p.getRight());
+                mplew.write(scriptableNpcIds.size());
+                scriptableNpcIds.forEach(id -> {
+                        mplew.writeInt(id);
+                        mplew.writeMapleAsciiString("NPC " + id); // The client needs a name for the npc, but it doesn't seem to do anything.
```
Comment (P0nk) [inline src/main/java/tools/MaplePacketCreator.java:8335]: I have no idea how the client treats this value, and I don't feel like testing it. LGTM.

**Review by P0nk — CHANGES_REQUESTED** [review]
Looks good overall. Certainly nice looking code by the existing standards.
I don't like the rebirth system personally, but I know others do, so I will keep it around, and it's easier to justify doing so with these changes.

File: src/main/java/net/server/channel/handlers/PlayerLoggedinHandler.java (hunk 1 above)
Comment (wejrox) [reply src/main/java/net/server/channel/handlers/PlayerLoggedinHandler.java]: Thanks, should be fixed.

**Comment by P0nk** [comment]
scriptable-npc-name
I found where the name is used. In this case I enabled Jade as the rebirth npc and made her scriptable. 
Displaying the NPC id like this is not very user friendly, so I say we should keep a name and send that to the client as the original implementation did. 

Suggestion: make `NPCS_SCRIPTABLE` a `Map` with id as key and name as value. Yamlbeans has map example here: https://github.com/EsotericSoftware/yamlbeans#basic-deserialization

**Comment by wejrox** [comment]
I've made the changes requested, but I couldn't get the yaml parser to accept an `Integer` as the key within `config.yaml` so I had to map the key to an `Integer` on use instead.

**Review by P0nk — APPROVED** [review]

**Pull request merged**